pysmt/smtlib/script.py

```diff
@@ -329,66 +333,129 @@ def smtlibscript_from_formula(formula, logic=None):
+            return solver.get_values(cmd.args)
 
-    elif cmd.name == smtcmd.EXIT:
-        return solver.exit()
+        elif cmd.name == smtcmd.PUSH:
+            return solver.push(cmd.args[0])
 
-    elif cmd.name == smtcmd.SET_LOGIC:
-        name = cmd.args[0]
-        return solver.set_logic(name)
+        elif cmd.name == smtcmd.POP:
+            return solver.pop(cmd.args[0])
 
-    elif cmd.name == smtcmd.DECLARE_FUN:
-        return solver.declare_fun(cmd.args[0])
+        elif cmd.name == smtcmd.EXIT:
+            return solver.exit()
 
-    elif cmd.name == smtcmd.DECLARE_CONST:
-        return solver.declare_const(cmd.args[0])
+        elif cmd.name == smtcmd.SET_LOGIC:
+            name = cmd.args[0]
+            return solver.set_logic(name)
 
-    elif cmd.name == smtcmd.DEFINE_FUN:
-        (var, formals, typename, body) = cmd.args
-        return solver.define_fun(var, formals, typename, body)
+        elif cmd.name == smtcmd.DECLARE_FUN:
+            return solver.declare_fun(cmd.args[0])
 
-    elif cmd.name == smtcmd.ECHO:
-        print(cmd.args[0])
-        return None
+        elif cmd.name == smtcmd.DECLARE_CONST:
+            return solver.declare_const(cmd.args[0])
 
-    elif cmd.name == smtcmd.CHECK_SAT_ASSUMING:
-        return solver.check_sat(cmd.args)
+        elif cmd.name == smtcmd.DEFINE_FUN:
+            (var, formals, typename, body) = cmd.args
+            return solver.define_fun(var, formals, typename, body)
 
-    elif cmd.name == smtcmd.GET_UNSAT_CORE:
-        return solver.get_unsat_core()
+        elif cmd.name == smtcmd.ECHO:
+            return cmd.args[0]
 
-    elif cmd.name == smtcmd.DECLARE_SORT:
-        name = cmd.args[0].name
-        arity = cmd.args[0].arity
-        return solver.declare_sort(name, arity)
+        elif cmd.name == smtcmd.CHECK_SAT_ASSUMING:
+            return solver.check_sat(cmd.args)
 
-    elif cmd.name in smtcmd.ALL_COMMANDS:
-        raise NotImplementedError("'%s' is a valid SMT-LIB command "\
-                                  "but it is currently not supported. "\
-                                  "Please open a bug-report." % cmd.name)
-    else:
-        raise UnknownSmtLibCommandError(cmd.name)
+        elif cmd.name == smtcmd.GET_UNSAT_CORE:
+            return solver.get_unsat_core()
+
+        elif cmd.name == smtcmd.DECLARE_SORT:
+            name = cmd.args[0].name
+            arity = cmd.args[0].arity
+            return solver.declare_sort(name, arity)
+
+        elif cmd.name in smtcmd.ALL_COMMANDS:
+            raise NotImplementedError("'%s' is a valid SMT-LIB command "\
+                                      "but it is currently not supported. "\
+                                      "Please open a bug-report." % cmd.name)
+        else:
+            raise UnknownSmtLibCommandError(cmd.name)
+
+
+class InterpreterOMT(InterpreterSMT):
+
+    def __init__(self):
+        self.optimization_goals = ([],[])
```

Comment:
Maybe add some comments about the class and how this field is used for future reference